### PR TITLE
Table: Fix wrong height when switch prop height.

### PR DESCRIPTION
**Pull request by liril-net**
Please makes sure these boxes are checked before submitting your PR, thank you!

* [x] Make sure you follow Element's contributing guide ([中文](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.zh-CN.md) | [English](https://github.com/ElemeFE/element/blob/master/.github/CONTRIBUTING.en-US.md)).
* [x] Make sure you are merging your commits to `dev` branch.
* [x] Add some descriptions and refer relative issues for you PR.

### CODE REVIEW

**Comment by liril-net** [comment]
#2949

File: packages/table/src/table.vue [hunk 1]
```diff
@@ -294,6 +295,22 @@
         return style;
       },
 
+      tableHeight() {
+        let style = {};
+
+        if (this.height) {
+          style = {
+            height: this.layout.tableHeight ? this.layout.tableHeight + 'px' : ''
```
Comment (QingWei-Li) [inline packages/table/src/table.vue]: 重复的代码是不是可以优化下

Comment (liril-net) [inline packages/table/src/table.vue]: 我复制的=。=，大佬！我改！

**Comment by liril-net** [comment]
@Leopoldthecoder CI 那个错误，我不会修

**Comment by Leopoldthecoder** [comment]
rebuild 一下就过了😂

**Comment by liril-net** [comment]
@Leopoldthecoder 但是那个应该是测试用例写的不完善吧？

**Comment by Leopoldthecoder** [comment]
@liril-net 应该是，之前也出现过

**Pull request merged**